aws-deadline/deadline-cloud-for-houdini
# fix: apply farm ID and queue ID settings correctly

### What was the problem/requirement? (What/Why)
The error message "AccessDeniedException" was appearing during login, settings, and creation flows with no actionable error message.

Resolves https://github.com/aws-deadline/deadline-cloud-for-houdini/issues/117

### What was the solution? (How)
Updated the flows to check whether the farm and queue IDs are set before calling Deadline Cloud APIs. Use a shared function for consistent behaviour.

Now, customers will only see error messages if they attempt to submit without a farm and queue ID set. The ROP creation, login, and settings flows will not show an error message if the farm and queue ID are not set.

### What is the impact of this change?
Avoids customer confusion from access denied exceptions

### How was this change tested?
1. Unset the farm ID and queue ID using `deadline config set defaults.farm_id ""` and `deadline config set defaults.queue_id`
2. Created a simple geometry, camera, and light scene with a Mantra ROP
3. Created a Deadline Cloud Node 
Before: access denied error appears
After: no error
4. Clicked login
Before: access denied error appears, even though login succeeded
After: no error
3. Opened the settings menu, then clicked okay without setting the farm and queue
Before: access denied error appears, even though there was no action by the user
After: no error
5. Submitted the job. An error message that the farm ID was not set appeared
6. Set the farm ID using `deadline config set defaults.farm_id <farm ID>`
7. Created a Deadline Cloud Node
Before: access denied error appears
After: no error
8. Clicked login
Before: access denied error appears, even though login succeeded
After: no error
9. Opened the settings menu, then clicked okay without setting the farm and queue
Before: access denied error appears, even though there was no action by the user
After: no error
10. Submitted the job. An error message that the queue ID was not set appeared
11. Set the queue ID using `deadline config set defaults.queue_id <queue ID>`
12. Created a Deadline Cloud node, clicked login, opened settings, and submitted a job

### Was this change documented?
No, not required

### Is this a breaking change?
No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*
